experimental/Schemes/SpaceGerms.jl

```diff
@@ -283,21 +981,275 @@ function Base.union(X::AbsSpaceGerm, Y::AbsSpaceGerm)
   return Z
 end
 
+## note: union of hypersurface germs is again a hypersurface germ
+##       union of complete intersection germs need not even be equidimensional
+function Base.union(X::HypersurfaceGerm, Y::HypersurfaceGerm)
+  R = ambient_coordinate_ring(X)
+  R === ambient_coordinate_ring(Y) || error("not subgerms of a common space germ")
+  point(X) == point(Y) || error("not the same point of the germ")
+  # comparison of points implicitly also checks that localization was performed at points
+  # otherwise 'point' is not implemented
+  f_new = numerator(X.f)*numerator(Y.f)
+  f_new = radical(ideal(R,[f_new]))[1]
+  Y = HypersurfaceGerm(Spec(quo(R,ideal(R,f_new))[1]), point(X))
+  Y.f = f_new
+  return Y
+end
+
 ##############################################################################
 # note: singular_locus, is_smooth and is_regular are inherited from Spec
 ##############################################################################
 
-# We want the singular locus of a `SpaceGerm` to be a `SpaceGerm` again and 
+# We want the singular locus of an `AbsSpaceGerm` to be a `SpaceGerm` again and
 # not a plain `Spec`.
+@doc raw"""
+   singular_locus(X::AbsSpaceGerm)  --> SpaceGerm, ClosedEmbedding
+
+Returns the space germ (Y,p) for a given germ (X,p) and the closed embedding of (Y,p) into (X,p), where Y is the singular locus of X.
+
+# Examples:
+```jldoctest
+julia> X = affine_space(QQ,3);
+
+julia> R = coordinate_ring(X);
+
+julia> (x,y,z) = gens(R);
+
+julia> Q,_ = quo(R,ideal(R,[x^2+y^2]));
+
+julia> Y = SpaceGerm(Spec(Q),[0,0,0]);
+
+julia> XSL, incSL = singular_locus(Y);
+
+julia> XSL
+Spectrum
+  of localization
+    of quotient
+      of multivariate polynomial ring in 3 variables x1, x2, x3
+        over rational field
+      by ideal(x1^2 + x2^2, x2, x1)
+    at complement of maximal ideal of point (0, 0, 0)
+
+julia> incSL
+Morphism
+  from [x1, x2, x3]  Spec of localization of quotient of multivariate polynomial ring at complement of maximal ideal
+  to   [x1, x2, x3]  Spec of localization of quotient of multivariate polynomial ring at complement of maximal ideal
+given by the pullback function
+  x1 -> x1
+  x2 -> x2
+  x3 -> x3
+
+```
+"""
 function singular_locus(X::AbsSpaceGerm)
   S, inc = singular_locus(underlying_scheme(X))
   Sgerm = SpaceGerm(S)
   return Sgerm, ClosedEmbedding(SpecMor(Sgerm, X, pullback(inc), check=false), image_ideal(inc), check=false)
 end
 
-function subscheme(X::SpaceGerm, I::Ideal)
+## note: subgerms of hypersurface and complete intersection germs are simply space germs
+@doc raw"""
+    subgerm(X::AbsSpaceGerm, I::Ideal) --> SpaceGerm
+
+Returns the space germ (Y,p) of (X,p) defined by the ideal I in the local ring of X at p
+
+!!! note: (Y,p) is of type SpaceGerm, even if (X,p) is a HypersurfaceGerm or a CompleteIntersectionGerm.
+"""
+function subgerm(X::AbsSpaceGerm, I::Ideal)
```

Comment:
I think we should turn this into a method of `subscheme`. Otherwise people don't find it and are wondering why `subscheme` returns something else.

Reply:
Could we agree on defining subscheme here and aliasing subgerm to it? People thinking in terms of germs and having created germs on the basis of local rings, will never think of subscheme.

Reply:
Yes, that's a possibility. But `subscheme` should really do the same here and I'm afraid it might not.

Reply:
They are now aliased to one another.